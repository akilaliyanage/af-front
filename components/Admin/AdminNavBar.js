import React, {useEffect, useState} from 'react';
import '../../assets/css/admin/admin.css'
import {useHistory} from "react-router-dom";

function AdminNavbar(){

    const  history = useHistory();

    const [admin,setAdmin] = useState();
    const [visible,setVisible] = useState('none')


    useEffect(() => {
        const admin = localStorage.getItem('user_name');
        if(admin !== null){
            setAdmin(admin);
            setVisible('')
        }
    });

    const logout = () => {
        localStorage.clear();
        history.push("/admin/login");
    }


    return(
        <div>
            <nav className="navbar navbar-expand-lg navbar navbar-dark bg-dark">
                <a className="navbar-brand" href="/admin">Dashboard</a>
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav"
                        aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav">
                        <li className="nav-item active">
                            <a className="nav-link" href="/home">Home </a>
                        </li>
                        <li className="nav-item">
                            <a style={{display:visible }} onClick={logout} className="nav-link" href="#">Logout</a>
                        </li>
                        <li className="nav-item">
                            <a style={{display:visible }} className="nav-link" href="#">Logged in as {admin}</a>
                        </li>


                    </ul>
                </div>
            </nav>
        </div>
    )
}

export default AdminNavbar;