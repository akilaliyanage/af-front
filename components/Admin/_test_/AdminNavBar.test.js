import React  from 'react';
import {render, screen, cleanup} from '@testing-library/react'
import AdminNavBar from '../AdminNavBar'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import expectExport from 'expect';


afterEach(()=>cleanup())

it('Compare the snapshot of Admin Nav Bar', () =>{
  const tree = renderer.create(
      <AdminNavBar/>
  ).toJSON();
  expectExport(tree).toMatchSnapshot();
})


test('Render Admin Nav Bar', () => {
    render(<AdminNavBar />);
    screen.debug();
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
  });

