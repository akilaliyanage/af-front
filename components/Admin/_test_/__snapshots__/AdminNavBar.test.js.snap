// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare the snapshot of Admin Nav Bar 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      Dashboard
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="#"
          >
            Home 
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`create a snapshot of Admin Nav Bar 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      Dashboard
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="#"
          >
            Home 
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
